### PR TITLE
Fixes the production build and removes dev dependencies from it.

### PR DESCRIPTION
**Pull request by zinigor**
Fixes the production build command that was broken. Before you would have to install every dependency to build Jetpack, now only needed dependencies are required and installed on `yarn build-production`.

#### Changes proposed in this Pull Request:

* Fixed the production build CLI command.
* Removed development dependencies from the production build.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This is an attempt to fix the broken Beta Builder by reducing the number of binaries we build on `yarn build`.

#### Testing instructions:

* Try to `yarn build`, you should get a working Jetpack installation.
* Try to `yarn build-production`, you should get a working Jetpack installation with minified code and less installed dependencies. Some things will be broken - like linting, commit checks, etc, but these are not necessary for production builds.

#### Proposed changelog entry for your changes:

* Fixed the CLI command to build Jetpack in production mode.

### CODE REVIEW

**Comment by gravityrail** [comment]
Legendary. Thank you.

**Review by dereksmart — CHANGES_REQUESTED** [review]
Q: How did you find and decide on all the modules that you moved? 

Test results: 
- `git checkout fix/production-builds`
- `git pull`
- `git reset --hard origin/fix/production-builds` (just in case)
- `yarn build-production`
- ✅ "Done in 188.57s." (a bit faster than normal)
-  😕 I notice there are un-committed local changes now? `git status`
image
- ❗️ Load the site, I get `Your installation of Jetpack is incomplete. If you installed Jetpack from GitHub, please refer to this document to set up your development environment.`
- ❗️ Verified that `vendor/autoload_packages.php` is indeed missing. 
- 🤷‍♂ Manually running `composer install` after the build fixes it for me.

File: package.json [hunk 1]
```diff
@@ -28,8 +28,9 @@
 		"build-client": "gulp",
 		"build-extensions": "webpack --config ./webpack.config.extensions.js",
 		"build-php": "composer install",
+		"build-production-php": "COMPOSER_MIRROR_PATH_REPOS=1 composer install --no-dev --classmap-authoritative",
```
Comment (dereksmart) [inline package.json]: Let's `-o` [optimize](https://getcomposer.org/doc/articles/autoloader-optimization.md#trade-offs) here?

Comment (dereksmart) [inline package.json]: For some reason `--no-dev` is causing our custom `autoload_packages.php` file to not get created. Maybe just need to move autoloader package to `require`?

**Review by dereksmart — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (dereksmart) [reply package.json]: if we added to `require`, then we would also need to `.svnignore` it 

adding to `require` does work fwiw.

**Comment by dereksmart** [comment]
in 301e61d I did what I suggested :) 

The autoloader needs to be in `require` if we use `--no-dev`, because the output of that autoloader is a file that is needed during production. 

Also added the -o command. 

Seems to actually build a working plugin now with `yarn build-production`. Have not yet tested `yarn build`

**Review by dereksmart — DISMISSED** [review]

**Comment by zinigor** [comment]
I can't approve, but I have tested both production and dev builds, both work fine. Thanks for fixing, Derek!

**Review by dereksmart — APPROVED** [review]

**Pull request merged**